### ISSUE TITLE
MultiNodeClusterSpec atStartup infinite loop

### ISSUE THREAD

**Issue by raboof** [akka/akka#30689]
AFAICT the MutliNodeClusterSpec initialization is essentially like this:

```    
trait MultiNodeSpec {
  protected def atStartup(): Unit = {
  }

  def beforeAll(): Unit = {
    atStartup()
  }
}

trait MultiNodeClusterSpec {
  self: MultiNodeSpec =>

  override protected def atStartup(): Unit = {
    self.atStartup()
  }
}

class DownAllUnstable5NodeSpec extends MultiNodeSpec with MultiNodeClusterSpec

object Main extends App {
  new DownAllUnstable5NodeSpec().beforeAll()
}
```

On Scala 3, this goes into an infinite loop.

In Akka it seems to work fine on Scala 2.13.6, however, when I try the isolated version above it also produces an infinite loop. To investigate further.

**Comment by patriknw** [comment]
This might be fixed by https://github.com/akka/akka/pull/30886, I haven't tried.

**Comment by raboof** [comment]
> This might be fixed by #30886, I haven't tried.

I think it looks like that: `sbt -Dakka.build.scalaVersion=3.x -Dakka.test.multi-node=true akka-cluster/test` fails for me since I don't have multi-node testing enabled, but I think earlier it failed before it even got that far. Let's see if the nightly builds agree after merging #30890 :)

**Comment by patriknw** [comment]
It failed as multi-jvm, so you don't have to set `-Dakka.test.multi-node=true`.
`akka-cluster/test` or `akka-cluster/multi-jvm:test` should be enough

agreed, that we can look at the nightlies
